src/Bridge/Doctrine/Orm/Extension/OrderExtension.php

```diff
@@ -52,7 +52,9 @@ public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGenerator
                         $field = $order;
                         $order = 'ASC';
                     }
-                    if (false === ($pos = strpos($field, '.'))) {
+
+                    if (false === ($pos = \strpos($field, '.'))
+                        || $pos && \array_key_exists(\substr($field, 0, $pos), $classMetaData->embeddedClasses)) {
```

Comment:
Could `isset($classMetaData->embeddedClasses[substr($field, 0, $pos)])` work? I think it's faster.

Also, not sure that `$pos &&` is required because the position will never be 0:

```
if (
  false === ($pos = strpos($field, '.'))
  ||
  isset($classMetaData->embeddedClasses[substr($field, 0, $pos)])
) {

}
```

Reply:
- I'll use `isset` then ;-)
- In case that `$pos = false`, `substr($field, 0, $pos)` will throw an exception since `$pos` is a boolean. Add `$pos &&` ensures us that is not boolean

Reply:
Mhh, but because the left part of the condition would be `true` with `$pos` being `false` it will never test the second part of the condition:

```
> false === ($pos = false) || false
true
```

Reply:
Yeah, my bad ! `$pos &&` is now removed.